src/ethereum/state_oracle/interface.py

```diff
@@ -0,0 +1,213 @@
+"""
+Abstract interface for state oracle implementations.
+"""
+
+from typing import Any, Callable, Optional, Protocol
+
+from ethereum_types.bytes import Bytes20, Bytes32
+from ethereum_types.numeric import U256
+
+# Use generic types for compatibility across forks
+Account = Any
+Address = Bytes20
+
+
+class MerkleOracle(Protocol):
```

Comment:
Hm, so we have a vague rule that if something changes between forks, it must be placed into the fork's folder. For example, `keccak256` is used in every fork and isn't likely to disappear, so it lives in `ethereum.crypto`, where as `Address` might get bumped to 32 bytes in the future, so it's defined per fork.

If the MPT is going to get replaced (or is fairly likely), it might be best to define it per fork.

Reply:
related to https://github.com/ethereum/execution-specs/pull/1371#discussion_r2286680828